### PR TITLE
feat(virtual_machine): add support for managing group assignment

### PR DESCRIPTION
**Pull request by andrewjtait**
requires https://github.com/krystal/go-katapult/pull/79

Allows a virtual machine group to be assigned to a virtual machine.

### CODE REVIEW

File: internal/provider/resource_virtual_machine_test.go [hunk 1]
```diff
@@ -583,6 +592,105 @@ func TestAccKatapultVirtualMachine_update_ips(t *testing.T) {
...
+				}`,
+				),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKatapultVirtualMachineExists(
+						tt, "katapult_virtual_machine.base",
+					),
+					resource.TestCheckResourceAttr(
+						"katapult_virtual_machine.base",
+						"group_id", "",
+					),
```
Comment (andrewjtait) [inline internal/provider/resource_virtual_machine_test.go:687]: I'd prefer to use `resource.TestCheckNoResourceAttr` here but unfortunately it gets set to an empty string 🤔

Comment (jimeh) [reply internal/provider/resource_virtual_machine_test.go:687]: Hmm, I'm just guessing off the top of my head, but I think that might be fixable in the Read function when setting `group_id`:

```go
	if vm.Group != nil && vm.Group != katapult.NullVirtualMachineGroup {
 		_ = d.Set("group_id", vm.Group.ID)
 	}
```

Comment (andrewjtait) [reply internal/provider/resource_virtual_machine_test.go:687]: I'm not sure if `TestCheckNoResourceAttr` checks for the key, and I cannot find any way to delete a key. `Set` will only update the existing state, I think.

File: internal/provider/resource_virtual_machine_test.go [hunk 2]
```diff
@@ -583,6 +592,105 @@ func TestAccKatapultVirtualMachine_update_ips(t *testing.T) {
...
+				resource "katapult_virtual_machine" "base" {
+					package       = "rock-3"
+					disk_template = "ubuntu-18-04"
+					disk_template_options = {
+						install_agent = true
+					}
+					ip_address_ids = [
+						katapult_ip.web.id,
+					]
+				}`,
```
Comment (jimeh) [inline internal/provider/resource_virtual_machine_test.go:678]: Indentation is off by one here :)

```suggestion
					resource "katapult_ip" "web" {}

					resource "katapult_virtual_machine" "base" {
						package       = "rock-3"
						disk_template = "ubuntu-18-04"
						disk_template_options = {
							install_agent = true
						}
						ip_address_ids = [
							katapult_ip.web.id,
						]
					}`,
```

**Review by jimeh — CHANGES_REQUESTED** [review]

File: internal/provider/resource_virtual_machine_test.go (hunk 1 above)
Comment (jimeh) [reply internal/provider/resource_virtual_machine_test.go:689]: Yeah, reading the docs it's about ensuring no value exists. I think it's fine to just ensure the value is a empty string, as we, and Go, treats that as not set.

**Pull request merged**